core/src/main/java/org/apache/iceberg/Partitioning.java

```diff
@@ -177,4 +184,35 @@ public Void alwaysNull(int fieldId, String sourceName, int sourceId) {
       return null;
     }
   }
+
+  /**
+   * Builds a common partition type for all specs in a table.
+   *
+   * @param table a table with one or many specs
+   * @return the constructed common partition type
+   */
+  public static StructType partitionType(Table table) {
+    Map<Integer, NestedField> commonFields = Maps.newHashMap();
+
+    for (PartitionSpec spec : table.specs().values()) {
+      List<NestedField> fields = spec.partitionType().fields();
+
+      for (NestedField field : fields) {
+        int fieldId = field.fieldId();
+        NestedField existingField = commonFields.get(fieldId);
+
+        // partition fields may conflict in v1 tables
+        ValidationException.check(existingField == null || existingField.equals(field),
+            "Specs contain conflicting partition fields with ID %d",
```

Comment:
Could output the existing field here as well to illustrate the conflict?

Reply:
+1, can just use `"... with %s", field`

Reply:
Fixed.